### PR TITLE
feature: Violating agents, allocation reasons and more

### PR DESCRIPTION
**Pull request by joelbarmettlerUZH**

### CODE REVIEW

File: Visualizer/src/API/enums.js [hunk 1]
```diff
@@ -7,3 +7,9 @@ export const BlockerType = {
   STATIC: "static",
   DYNAMIC: "dynamic",
 };
+
+export const BRANCH_REASONS = {
+  REALLOCATION: "AGENT",
```
Comment (thommann) [inline Visualizer/src/API/enums.js]:
```suggestion
  REALLOCATION: "REALLOCATION",
```

File: Visualizer/src/components/dashboard/AgentInfo.vue [hunk 2]
```diff
@@ -239,7 +245,7 @@ const allocations = computed(() =>
       {
         label: "Collisions",
         value: stat.collidingAgentIds.length,
-        icon: GitPullRequest,
+        icon: Skull,
```
Comment (thommann) [inline Visualizer/src/components/dashboard/AgentInfo.vue:249]: Collisions are not the same as violations.

File: Visualizer/src/components/dashboard/TopBar.vue [hunk 3]
```diff
@@ -97,7 +89,7 @@ const stats = computed(() => {
     {
       label: "Collisions",
       value: simulation.statistics.totalNumberOfCollisions,
```
Comment (thommann) [inline Visualizer/src/components/dashboard/TopBar.vue:91]:
```suggestion
      label: "Violations",
      value: simulation.statistics.totalNumberOfCollisions,
```

Comment (thommann) [reply Visualizer/src/components/dashboard/TopBar.vue:91]: rename property as well. Should be called violations.

**Review by thommann — CHANGES_REQUESTED** [review]

File: Visualizer/src/components/dashboard/TopBar.vue (hunk 3 above)
Comment (joelbarmettlerUZH) [reply Visualizer/src/components/dashboard/TopBar.vue:91]: I don't get it. The API property is called "total_number_of_collisions", I name it "totalNumberOfCollisions" and display it under the title "Collisions", but you are telling me it should be called Violations?

Comment (thommann) [reply Visualizer/src/components/dashboard/TopBar.vue:91]: yes

Comment (thommann) [reply Visualizer/src/components/dashboard/TopBar.vue:91]: habs in meinem PR gefixed

File: Visualizer/src/components/dashboard/AgentInfo.vue [hunk 4]
```diff
@@ -275,12 +282,36 @@ const allocations = computed(() =>
...
+        label: `${agent.owner.name} > ${agent.id}`,
+        value: { x: violation.x, y: violation.y, z: violation.z },
+        icon: GitPullRequest,
+        onClick: () => {
+          simulation.focusOnAgent(agent);
+        },
+      });
+      return acc;
+    }, []);
 });
```
Comment (johannschwabe) [inline Visualizer/src/components/dashboard/AgentInfo.vue:316]: cyclic complexity to hight

**Review by johannschwabe — COMMENTED** [review]
I'm proud of you. But it's to complex

**Pull request closed without merging**